app/assets/javascripts/oxalis/view/version_list.js

```diff
@@ -83,28 +102,71 @@ class VersionList extends React.Component<Props, State> {
     Store.dispatch(setAnnotationAllowUpdateAction(true));
   };
 
+  previewVersion = (version: number) => previewVersion({ [this.props.tracingType]: version });
+
+  getGroupedAndChunkedVersions = _.memoize(
+    (versions: Array<APIUpdateActionBatch>): GroupedAndChunkedVersions => {
+      // This function first groups the versions by day, where the key is the output of the moment calendar function.
+      // Then, the versions for each day are chunked into x-minute intervals,
+      // so that the actions of one chunk are all from within one x-minute interval.
+      const groupedVersions = _.groupBy(versions, batch =>
+        moment
+          .utc(_.max(batch.value.map(action => action.value.actionTimestamp)))
+          .calendar(null, MOMENT_CALENDAR_FORMAT),
+      );
+
+      const CHUNK_BY_X_MINUTES = 5;
+      return _.mapValues(groupedVersions, versionsOfOneDay => {
+        let chunkIndex = 0;
+        let chunkTime = 0;
+        return _.reduce(
```

Comment:
Do you think this can be refactored into a helper function, which can be used like:

```
chunkIntoTimeWindows(versionsOfOneDay, batch => _.max(batch....), CHUNK_BY_X_MINUTES)
```

First parameter would be the data array, second parameter the function which maps an entry to a time and the third the window size? Then, we could also write a unit test for the chunking code :)

Reply:
What a marvelous idea :grinning: I did both of that :)